angr/analyses/xrefs.py

```diff
@@ -109,7 +117,12 @@ class XRefsAnalysis(ForwardAnalysis, Analysis):  # pylint:disable=abstract-metho
         23ca - read access
         23ce - write access
     """
-    def __init__(self, func=None, func_graph=None, block=None, max_iterations=1, replacements=None):
+    def __init__(self, func=None, func_graph=None, block=None, max_iterations=1, replacements=None, start_state=None):
```

Comment:
Why do we need `start_state` at all? It does not seem to be used anywhere...

Reply:
Yes it is! It's used by the SimEngineLightXRefs or whatever to deal with the indirection

Reply:
Where is it used? Sorry that I could not find it.

Reply:
lol it gets assigned to self._start_state right below, and passed to process() on line 179, and that all gets used in do_load() on line 68.

Reply:
I think a better way is to perform the read in propagator (which I think it already does right now) and only get the read result in `XRefsAnalysis` -- the read result will always be in the lhs tmp. This way we can avoid having to re-read when running `XRefsAnalysis` and make it run faster!

Also, as a general rule of thumb, we should avoid using `SimState`s as much as we can in static analysis, because it's slow. We have methods in CLE to perform fast reads directly from the memory backers.

Reply:
ha, I was copying how you did it in propagator! :P

Reply:
Exactly... if we are doing it in propagator already, let's stick to that and don't do the read in xrefs again.

I should fix how the memory loading works in propagator, too.

Reply:
Oh, lol. The memory loading in propagator is a hack, which is why it is controlled by a callback `self._allow_loading()`.

I'm OK with the current design (having memory loads inside XRefsAnalysis) if you switch to the fast memory load and get rid of the state: see `_fast_memory_load_pointer()` in `cfg_base.py`.

Sorry for being picky, but speed matters!